packages/merge-styles/src/Stylesheet.ts

```diff
@@ -158,28 +159,28 @@ export class Stylesheet {
   public insertRule(
     rule: string
   ): void {
-    const element = this._getElement();
-    const injectionMode = element ? this._config.injectionMode : InjectionMode.none;
-
-    switch (injectionMode) {
-      case InjectionMode.insertNode:
-        const { sheet } = element!;
-
-        try {
-          // tslint:disable-next-line:no-any
-          (sheet as any).insertRule(rule, (sheet as any).cssRules.length);
-        } catch (e) {
-          /* no-op on errors */
-        }
-        break;
-
-      case InjectionMode.appendChild:
-        _createStyleElement(rule);
-        break;
-
-      default:
-        this._rules.push(rule);
-        break;
+    const { injectionMode } = this._config;
+    const element = injectionMode !== InjectionMode.none ? this._getStyleElement() : undefined;
```

Comment:
Nit: this nesting of a switch inside an if-else is not very readable to me. Any reason not to just have a switch covering all states?

Reply:
It was because the "else" statement applies to all states. If there is no element created, because say, `document` is not available, cache the rules in the array for later use.